app/src/main/kotlin/com/hippo/ehviewer/ui/main/AvatarIcon.kt

```diff
@@ -0,0 +1,189 @@
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.unit.dp
+import arrow.core.Either
+import arrow.core.Either.Companion.catch
+import arrow.core.Option
+import arrow.core.none
+import arrow.core.some
+import coil3.compose.AsyncImage
+import com.hippo.ehviewer.R
+import com.hippo.ehviewer.Settings
+import com.hippo.ehviewer.client.EhEngine
+import com.hippo.ehviewer.client.EhUtils
+import com.hippo.ehviewer.client.parser.HomeParser
+import com.hippo.ehviewer.collectAsState
+import com.hippo.ehviewer.ui.tools.DialogState
+import com.hippo.ehviewer.util.displayString
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.SharingStarted
+import kotlinx.coroutines.flow.StateFlow
+import kotlinx.coroutines.flow.conflate
+import kotlinx.coroutines.flow.map
+import kotlinx.coroutines.flow.merge
+import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.launch
+import moe.tarsin.coroutines.runSwallowingWithUI
+
+private val limitScope = CoroutineScope(Dispatchers.IO)
+private val refreshEvent = MutableSharedFlow<Unit>()
+private val invalidateEvent = MutableSharedFlow<Unit>()
+
+typealias Result = Option<Either<String, HomeParser.Result>>
+
+private val limitFlow: StateFlow<Result> = refreshEvent.conflate()
+    .onEach {
+        // If cached value is error, drop it when refresh
+        val isError = limitFlow.value.isSome { it.isLeft() }
+        if (isError) invalidateEvent.emit(Unit)
+    }
+    .map { catch { EhEngine.getImageLimits() }.mapLeft { e -> e.displayString() }.some() }
+    .let { src -> merge(src, invalidateEvent.map { none() }) }
+    .stateIn(limitScope, SharingStarted.Eagerly, none())
+
+context(CoroutineScope, DialogState, SnackbarHostState, DestinationsNavigator)
+@Composable
+fun AvatarIcon() {
+    val hasSignedIn by Settings.hasSignedIn.collectAsState()
+    if (hasSignedIn) {
+        val placeholder = stringResource(id = R.string.please_wait)
+        val resetImageLimitSucceed = stringResource(id = R.string.reset_limits_succeed)
+        val result by limitFlow.collectAsState()
+        IconButton(
+            onClick = {
+                launch {
+                    refreshEvent.emit(Unit)
+                    awaitConfirmationOrCancel(
+                        confirmButton = {
+                            TextButton(onClick = it, enabled = result.isSome { it.isRight { it.limits.resetCost != 0 } }) {
+                                Text(text = stringResource(id = R.string.reset))
+                            }
```

Comment:
Resetting image limit should not close the dialog as the dialog is the only place to show image limit now.

Reply:
Then should we show snackbar?

Reply:
Only on failure to show the reason? The user can tell from the new image limit if it succeeded.

Reply:
Done